### ISSUE TITLE
Results are not reproducible

### ISSUE THREAD

**Issue by aurotripathy** [huggingface/diffusers#1185]
### Describe the bug

Disappointed that the results are not reproducible (others way the same as well).
In my case, the output image is very similar to one of the input image.

My steps:
```
pip install --upgrade diffusers[torch]
git clone https://github.com/huggingface/diffusers.git
```
```
 cd diffusers/examples/dreambooth/
 pip install -U -r requirements.txt
 accelerate config
 huggingface-cli login
```

Get the seeding dog images as instructed

install `bitsandbytes`
```
pip install bitsandbytes
python -m bitsandbytes
export LD_LIBRARY_PATH="/opt/conda/lib"
python -m bitsandbytes
```
Run the finetune script

```
export MODEL_NAME="CompVis/stable-diffusion-v1-4"
export INSTANCE_DIR="/root/images/dog/"
export CLASS_DIR="path-to-class-images"
export OUTPUT_DIR="path-to-save-model"

accelerate launch train_dreambooth.py \
  --pretrained_model_name_or_path=$MODEL_NAME  \
  --train_text_encoder \
  --instance_data_dir=$INSTANCE_DIR \
  --class_data_dir=$CLASS_DIR \
  --output_dir=$OUTPUT_DIR \
  --with_prior_preservation --prior_loss_weight=1.0 \
  --instance_prompt="a photo of sks dog" \
  --class_prompt="a photo of dog" \
  --resolution=512 \
  --train_batch_size=1 \
  --use_8bit_adam \
  --gradient_checkpointing \
  --learning_rate=2e-6 \
  --lr_scheduler="constant" \
  --lr_warmup_steps=0 \
  --num_class_images=200 \
  --max_train_steps=800

```
Run the inference script

```
from diffusers import StableDiffusionPipeline
import torch

model_id = "path-to-save-model"
pipe = StableDiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float16).to("cuda")

prompt = "A photo of sks dog in a bucket"
image = pipe(prompt, num_inference_steps=50, guidance_scale=7.5).images[0]

image.save("dog-bucket.png")

```

### Logs

```shell
--
```

### System Info

--

**Comment by Revist** [comment]
Duplicate of https://github.com/huggingface/diffusers/issues/1062

**Comment by patil-suraj** [comment]
Answered here https://github.com/huggingface/diffusers/issues/1062#issuecomment-1307128786

**Comment by aurotripathy** [comment]
@patil-suraj, @Revist:
Thanks for the guidance. 
If I use _just_ the [prior-preservation](https://github.com/huggingface/diffusers/tree/main/examples/dreambooth#training-with-prior-preservation-loss) script
 I'm able to reproduce the result (something close). 

![new_1](https://user-images.githubusercontent.com/1825684/200697157-014a1faa-c736-4c1a-a22b-11dbef50f60a.png)

If I extend the fine-tuning to the `text-encoder`, the input is the same as the output. 

Is that expected?

I would love an answer here but I can close this and open a new one, it thats better.

Thank you again.

-Auro

**Comment by aurotripathy** [comment]
@entrpn, thank you. 
Would you be willing to share your setting for fine-tuning the `text encoder`(which ones)?
That would be helpful.

**Comment by patil-suraj** [comment]
@aurotripathy as I said in the comment I linked, for dream booth we need to tune hyperparams to get the best results. Also, training text encoder is usually helpful for more culex concepts like faces. I don't think this is an issue with training script. I would suggest to try different setting and pick what works best for your use-case.

**Comment by aurotripathy** [comment]
Thank you @patil-suraj 
Lots of good thoughts (for a newbie, me)
Closing this.

**Comment by camenduru** [comment]
> Would you be willing to share your setting for fine-tuning the text encoder(which ones)?
That would be helpful.

@aurotripathy maybe helps https://wandb.ai/psuraj/dreambooth/reports/Dreambooth-Training-Analysis--VmlldzoyNzk0NDc3

**Comment by entrpn** [comment]
@aurotripathy take a look at this project. https://github.com/entrpn/serving-model-cards/tree/main/training-dreambooth. That's what I used for the results above.
